test_data/asm/wrap_gl.asm

```diff
@@ -0,0 +1,123 @@
+// Wraps an arbitrary field element into a u32, on the Goldilocks field.
+machine WrapGL(RESET, operation_id) {
+
+    operation wrap<0> in_acc -> output;
+
+    constraints {
+
+        // Latch and operation ID
+        col fixed RESET(i) { ((i % 8) - 7) == 0 };
+        col witness operation_id;
+
+        // 1. Decompose the input into bytes
+
+        // The byte decomposition of the input, in little-endian order
+        // and shifted forward by one (to use the last row of the
+        // previous block)
+        col witness bytes;
+        // Puts the bytes together to form the input
+        col witness in_acc;
+        // Factors to multiply the bytes with
+        col fixed FACTOR(i) { 1 << (((i + 1) % 8) * 8) };
+
+        in_acc' = in_acc * (1 - RESET) + bytes * FACTOR;
```

Comment:
More of a question, but in conditional cases don't we usually write the condition first?
So here `(1 - RESET) * in_acc` like `!RESET ? in_acc : 0`